### ISSUE TITLE
New-PSSession from Windows to Linux fails due to ssh banner

### ISSUE THREAD

**Issue by thukk** [PowerShell/PowerShell#15759]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Set up Windows 10 and Linux host for SSH.
Create a New-PSSession from Windows to Linux using SSH transport.
`$Session = New-PSSession -HostName $ComputerName -UserName $SSHUsername -SSHTransport`

Prior to KB5003173 for 20H2, the error could be negated with error handling.  After that patch, the error is terminating regardless of how it is handled.

### Expected behavior

```console
New session should be created.
```

### Actual behavior

```console
The SSH banner is evaluated to stderr and the connection fails.
```

### Error details

```console
> get-error

Exception             :
    Type        : System.Management.Automation.Remoting.PSRemotingTransportException
    ErrorCode   : 2100
    ErrorRecord :
        Exception             :
            Type    : System.Management.Automation.ParentContainsErrorRecordException
            Message : The background process reported an error with the following message: The SSH client session has
ended with error message: "SSH Banner Test".
            HResult : -2146233087
        CategoryInfo          : ResourceUnavailable: (:) [], ParentContainsErrorRecordException
        FullyQualifiedErrorId : System.Management.Automation.Remoting.PSRemotingDataStructureException
    Message     : The background process reported an error with the following message: The SSH client session has
ended with error message: "SSH Banner Test".
    HResult     : -2146233087
TargetObject          : System.Management.Automation.RemoteRunspace
CategoryInfo          : OpenError: (System.Management.A…tion.RemoteRunspace:RemoteRunspace) [New-PSSession],
PSRemotingTransportException
FullyQualifiedErrorId : 2100,PSSessionOpenFailed
ErrorDetails          : [Linux_TestHost] The background process reported an error with the following message: The SSH
client session has ended with error message: "SSH Banner Test".
InvocationInfo        :
    MyCommand        : New-PSSession
    ScriptLineNumber : 755
    OffsetInLine     : 36
    HistoryId        : 3
    ScriptName       : C:\Users\username\Desktop\SSH_test.ps1
    Line             : $Session = New-PSSession -HostName $ComputerName -UserName $SSHUsername -SSHTransport
-ErrorAction Stop

    PositionMessage  : At C:\Users\username\Desktop\SSH_test.ps1
                       + …  $Session = New-PSSession -HostName $ComputerName -UserName $SSHUsern …
                       +               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    PSScriptRoot     : C:\Users\username\Desktop\
    PSCommandPath    : C:\Users\username\Desktop\SSH_test.ps1
    InvocationName   : New-PSSession
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock><Process>, C:\Users\username\Desktop\SSH_test.ps1:
line 755
                        at <ScriptBlock>, <No file>: line 1
```

### Environment data

```powershell
> $psversiontable

Name                           Value
----                           -----
PSVersion                      7.1.3
PSEdition                      Core
GitCommitId                    7.1.3
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by rkeithhill** [comment]
Can you try enabling the following experimental feature:

```powershell
Enable-ExperimentalFeature PSNotApplyErrorActionToStderr
```

Try doing that on the Linux side first to see if that resolves it.  If it doesn't, then try it on the client side.

**Comment by thukk** [comment]
no change.  Ran it on the Linux host as well as the Windows client.

both times, I closed and restarted pwsh before attempting to create a new session.

**Comment by thukk** [comment]
With `-EA SilentlyContinue`, I get "Error: Cannot validate argument on parameter 'Session'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again."

With `-EA Stop`, I get the SSH banner message as an error.

These errors come up when PS prompts for the SSH password - before I enter anything.

**Comment by iSazonov** [comment]
I don't understand how it is related to KB5003173.

**Comment by iSazonov** [comment]
Is this the same as #5142?

**Comment by thukk** [comment]
Yes - same as #5142.  The KB is relevant as before that KB, I could manage the error and get past it.  After that KB was installed, I get a terminating error regardless of how I manage the error actions.  There were some changes to the scripting engine that could have affected it.

As stated in 5142, this is quite a big deal in trying to secure environments.  SSH banners are a requirement is many areas and PS 7 not being to handle them is quite concerning.

**Comment by iSazonov** [comment]
@PaulHigin Do you prefer close this issue as duplicate and continue tracking in #5142?

**Comment by PaulHigin** [comment]
@iSazonov Thanks, yes this looks like a duplicate.

**Comment by thukk** [comment]
Respectfully, 5142 has not gotten updated since January.  This thread at least got a response to attempt an experimental feature...Is this an active bug/improvement being worked?

**Comment by PaulHigin** [comment]
@PowerShell/wg-powershell-remoting 

We can bump up the priority on this (#5142), but SSH writes the banner in the error stream, it is difficult to distinguish it from a legit error.  I think the solution will end up being fairly involved.

**Comment by thukk** [comment]
Thank you very much.

**Comment by JacobStruiksma** [comment]
So  `Enable-ExperimentalFeature PSNotApplyErrorActionToStderr` does nothing at all.  Stil does not work for 7.1.4 or 7.2.0-preview.8.

Isn't kind of the same issue with az cli and stderr stream? was going to update #5142 but saw this issue also.

**Comment by thukk** [comment]
Any update on progress with this bug?  This issue also happens when attempting to created a connection from Linux to Linux.  I can bypass the SSH prompt by setting no banner in sssh_config, but if FIPs Mode is enabled, the connection errors out as SSH says "The background process reported an error with the following message: The SSH client session has ended with error message: FIPS mode initialized."

Even if it writes to `stderr`, that is fine - but at least make it respect the ErrorAction setting (it currently Stops, regardless of the set Error Action).  That would be bare minimum.

**Comment by SteveL-MSFT** [comment]
This was already reviewed by @PowerShell/powershell-committee to have as experimental feature in 7.3

**Comment by daxian-dbw** [comment]
Close via https://github.com/PowerShell/PowerShell/pull/16205

**Comment by thukk** [comment]
I just pulled down 7.3 Preview 1, ran `Enable-ExperimentalFeature PSRemotingSSHTransportErrorHandling` and viola!  with `-EA Continue`, I was able to see the banner and still enter my SSH password to create a valid session.  `SilentlyContinue` hides the basement completely, which is also nice.  

If I had one complaint, the SSH banner shows at the password prompt, so the cursor is awaiting input on an empty line - but that is a somewhat minor inconvenience.

Thank you for getting this working.
